nevis/_interpolation.py

```diff
@@ -67,6 +67,27 @@ def __call__(self, x, y):
         # Final result
         return np.where(f1 == f2, f1, (y2 - y) * f1 + (y - y1) * f2)
 
+    def grad(self, x, y):
```

Comment:
Should we make it so that this returns the value _and_ the gradient? I can't think of many scenarios where you just want the gradient, and the functions overlap quite a bit

Reply:
Could you also explain the returned format in the docstring? Maybe give an example too?

Reply:
I see... So is it better (and what you originally meant) to incorporate `grad` as an optional parameter of the function, which makes the function return extra fields when set `True`?

Reply:
You mean like `f(x, y, True)` would give you gradients?
Yeah we could do that!
Or I suppose we could write an entirely separate function (or callable class) ?
Up to you!

Reply:
In PINTS we have a `__call__` for just f, and then a second function for f+df/dx: https://github.com/pints-team/pints/blob/master/pints/_error_measures.py#L12-L42

There's probably no right or wrong, just pick whatever you think is easiest!

Reply:
To minimize code duplication here, I chose to use an optional parameter. Please have a check!

Reply:
At the three points the function returns `(array(-17.86381531), (0.000200042724609375, 0.000200042724609375))`.

Reply:
BTW, do we need to make the function value a float?

Reply:
What about just outside, e.g. (-1, 0) ?

Reply:
> BTW, do we need to make the function value a float?

Not sure what you mean? Python 3 does this automatically if needed

Reply:
Like `-17.86381531` instead of `array(-17.86381531)`. I encountered an error causes by this type issue when using NLopt (which is not Python implemented).

Reply:
> What about just outside, e.g. (-1, 0) ?

Same result :-)

Reply:
Aha! Hadn't noticed that! Yeah make it a float